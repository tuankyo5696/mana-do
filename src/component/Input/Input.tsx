import React, { ChangeEvent, CSSProperties, InputHTMLAttributes } from 'react';

interface InputProps extends InputHTMLAttributes<HTMLInputElement> {
  type?: string;
  className?: string;
  textLabel?: string;
  isHasLabel?: boolean;
  onChange?: (e: ChangeEvent<HTMLInputElement>) => void;
  style?: CSSProperties;
}
const Input = React.forwardRef<HTMLInputElement, InputProps>(
  ({ id, isHasLabel, textLabel, name, type, value, style, onChange, ...args }, ref) => (
    <>
      {isHasLabel ? <label htmlFor={id}>{textLabel}</label> : ''}
      <input
        id={id}
        name={name}
        type={type}
        style={{ ...style }}
        value={value}
        onChange={onChange}
        ref={ref}
        {...args}
      />
    </>
  )
);
Input.defaultProps = {
  onChange: () => {},
  style: {},
  isHasLabel: false,
  className: '',
  textLabel: '',
  type: 'text'
};

export default Input;
