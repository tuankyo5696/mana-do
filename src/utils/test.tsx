import React from 'react';
import renderer from 'react-test-renderer';

interface Config {
  props: object;
  description: string;
}

const singleSnapTest = (tree: any, description: string) => {
  test(description, () => {
    expect(tree).toMatchSnapshot();
  });
};

const testSnapshots = (Component: any, configs: Config[]) => {
  describe('snapshots', () => {
    configs.forEach((config) => {
      const { props, description } = config;
      const tree = renderer.create(<Component {...props} />).toJSON() 
      singleSnapTest(tree, description);
    });
  });
};

export { testSnapshots };
