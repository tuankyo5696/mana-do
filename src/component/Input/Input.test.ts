import { testSnapshots } from '../../utils/test';
import Input from './Input';
describe('render Input', () => {
  testSnapshots(Input, [
    {
      props: {
        isHasLabel: true,
        textLabel: 'password'
      },
      description: 'render input has label'
    },
    {
      props: {
        isHasLabel: false,
        textLabel: 'password'
      },
      description: 'render input does not have label'
    }
  ]);
});
