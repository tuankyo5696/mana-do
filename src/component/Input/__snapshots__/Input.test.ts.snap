// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Input snapshots render input does not have label 1`] = `
Array [
  "",
  <input
    className=""
    onChange={[Function]}
    style={Object {}}
    type="text"
  />,
]
`;

exports[`render Input snapshots render input has label 1`] = `
Array [
  <label>
    password
  </label>,
  <input
    className=""
    onChange={[Function]}
    style={Object {}}
    type="text"
  />,
]
`;
